scripts/shell.sh

```diff
@@ -97,6 +97,43 @@ preflightCheck() {
   esac
 }
 
+verify_nix_daemon_config_compatibility() {
+  # Check if there is no discrepancies between host nix daemon configuration
+  # and project disruptor requirements.
+  # 
+  # As one cannot assume the commands will be executed by a trusted nix user,
+  # only verification remains to notify user of any potential issues in host
+  # nix configuration.
+  #
+  # This function assumes nix is installed globally and present in $PATH.
+
+  local readonly config_difference;
+  local readonly global_nix_conf_settings="${CACHE_ROOT}/.global-nix-conf"
+
+  # `nix show-config` seems to be the only available way of obtaning 
+  # "final" nix configuration, without doing the re-implementation of 
+  # said logic. 
+  # Therefore "experimental-features = nix-command" are needed.
+
+  mkdir -p "${CACHE_ROOT}"
+  # Why not $(nix show-config)? It fails with cryptic 'access-tokens: command not found'
+  nix show-config --json > "${global_nix_conf_settings}" 2>/dev/null
+  # shellcheck disable=SC2181
+  if [ $? -ne 0 ]; then
+   fail "nix show-config command has failed. Please enable it by following this steps: https://nixos.wiki/wiki/Nix_command"
```

Comment:
Why not add `--experimental-features nix-command` to the command above?

Also this condition can be reached in cases when installed nix version is below 2.4.